src/Functions/Public/properties-service/New-vRAPropertyGroup.ps1

```diff
@@ -0,0 +1,208 @@
+﻿function New-vRAPropertyGroup {
+    <#
+    .SYNOPSIS
+    Create a custom Property Group
+    
+    .DESCRIPTION
+    Create a custom Property Group
+
+    .PARAMETER Name
+    The unique name (ID) of the Property
+    
+    .PARAMETER Label
+    The text to display in forms for the Property
+
+    .PARAMETER Description
+    Description of the Property
+
+    .PARAMETER Tenant
+    The tenant in which to create the Property Group (Defaults to the connection tenant )
+
+    .PARAMETER Properties
+    The tenant in which to create the Property Group (Defaults to the connection tenant )
+   
+    .PARAMETER JSON
+    Property Group to send in JSON format
+
+    .INPUTS
+    System.String.
+
+    .OUTPUTS
+    System.Management.Automation.PSObject
+
+    .EXAMPLE
+    # Create a simple property group with no properties addded
+    New-vRAPropertyGroup -Name one 
+    
+    .EXAMPLE
+    # Create a property group with a description and label
+    New-vRAPropertyGroup -Name OneWithDescription -Label "On With Description" -Description "This is one with a label and description"
+
+    .EXAMPLE
+    # Create a property group with some properties added in simple form
+    New-vRAPropertyGroup -Name OneWithPropetiesSimple -Label "One With Properties" -Properties @{"com.org.bool"=$false; "com.org.string"="string1"}
+
+    .EXAMPLE
+    # Create a property group with some properties added in the extended form
+    New-vRAPropertyGroup -Name OneWithPropertiesExt -Label "One With Properties" -Properties @{"com.org.bool"=@{"mandatory"=$true; "defaultValue"=$false;}; "com.org.encryptedandshowonform"=@{"encrypted"=$true; "visibility"=$true; "defaultValue": "Un-encrypted string";};}
+
+#> 
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = "Low")][OutputType('System.Management.Automation.PSObject')]
```

Comment:
Is there a reason you didn't specify a DefaultParameterSetName? If not, can you add one. I currently get this if not using a parameter from one of the two sets:
```
New-vRAPropertyGroup -Name OneWithDescription -Label "On With Description" -Description "This is one with a label and description"
New-vRAPropertyGroup : Parameter set cannot be resolved using the specified named parameters.
At line:1 char:1
+ New-vRAPropertyGroup -Name OneWithDescription -Label "On With Descrip ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-vRAPropertyGroup], ParameterBindingException
    + FullyQualifiedErrorId : AmbiguousParameterSet,New-vRAPropertyGroup
```

In fact I think you might need an additional parameter set to cater for the above example where neither Properties, nor JSON is specified as a parameter, or make some of them available in multiple sets.

Reply:
ok, set a Default ParameterSet to just include the Name property, and set it as the default set. Uploading soon.